experimental/GModule/GModule.jl

```diff
@@ -207,8 +212,71 @@ where `phi` is an embedding of `S` into `R`.
 
 (case that `R` is a number field and `S` is the ring of integers in `R`)
 """
-function invariant_lattice_classes(M::GModule, phi::Map)
-  error("not yet ...")
+function invariant_lattice_classes(M::GModule{<:Oscar.GAPGroup, <:AbstractAlgebra.FPModule{QQFieldElem}}, phi::Map{ZZRing, QQField})
+  MZ = gmodule(ZZ, M)
+  return invariant_lattice_classes(MZ)
+end
+
+function Oscar.pseudo_inv(h::Generic.ModuleHomomorphism)
+  return MapFromFunc(codomain(h), domain(h), x->preimage(h, x))
+end
+
+function _hom(f::Map{<:AbstractAlgebra.FPModule{T}, <:AbstractAlgebra.FPModule{T}}) where T
+  @assert base_ring(domain(f)) == base_ring(codomain(f))
+  return hom(domain(f), codomain(f), f.(gens(domain(f))))
+end
+
+function invariant_lattice_classes(M::GModule{<:Oscar.GAPGroup, <:AbstractAlgebra.FPModule{ZZRingElem}})
+  res = Any[(M, sub(M.M, gens(M.M))[2])]
+  sres = 1
+  new = true
+  lp = keys(factor(order(M.G)).fac)
+  while new
+    new  = false
+    lres = length(res)
+    for X = res[sres:end]
+      for p = lp
+        F = free_module(GF(p), dim(M))
+        pM = gmodule(M.G, [hom(F, F, map_entries(base_ring(F), x)) for x = map(matrix, action(M))])
+        S = maximal_submodule_bases(pM)
+        pG = p.*gens(M.M)
+        for s = S
+          x, mx = sub(M.M, vcat(pG, [M.M(map_entries(x->lift(ZZ, x), s[i:i, :])) for i=1:nrows(s)]))
+
+          r = (gmodule(M.G, [_hom(mx*h*pseudo_inv(mx)) for h = M.ac]), mx)
+          if any(x->is_isomorphic(r[1], x[1]), res)
+            continue
+          else
+            new = true
+            push!(res, r)
+          end
+        end
+      end
+    end
+    lres = length(res)
```

Comment:
Redundant?
```suggestion
```
Also, with this gone, you could also get rid of variable `new` and instead do at the top:
```
  lres = 0
  while lres < length(res)
    lres = length(res)
```

Reply:
This was Claus' code, I'm supposed to be looking it over to understand and see where improvements can be made; it seems like this change removing `new` leaves everything functioning the same, but I'm double checking everything.